### ISSUE TITLE
Volumetric light wont get blocked by objects when shadows are enabled

### ISSUE THREAD

**Issue by Xyotic** [godotengine/godot#66320]
### Godot version

Godot4 Beta1

### System information

Windows10, Vulkan, RTX2080

### Issue description

Enabling volumetrics in WorldEnvironment and on a light source, the volumetric light passes through objects even if shadows are enabled.
![grafik](https://user-images.githubusercontent.com/8405059/192051405-1f326358-6a68-40f0-8045-9890622b5e38.png)
![grafik](https://user-images.githubusercontent.com/8405059/192065146-9cf4ff81-8b0b-49be-80e1-952df3f74371.png)

I've attached this scene as a mrp.

Edit: it seems to be dependent on the viewed angle... its most noticable when looking directly at the back face of the box.

### Steps to reproduce

- Create a world evnironment with volumetric fog enabled
- Add a light source with volumentric fog energy > 0
- Enable shadows on light source
- Put an obstacle infront of the light source

See mrp

### Minimal reproduction project
[Test_4_0.zip](https://github.com/godotengine/godot/files/9636979/Test_4_0.zip)

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/57722 (same cause – volumetric fog shadow rendering is broken for SpotLights in general).

**Comment by Xyotic** [comment]
"closed as not planned"?
Does this mean we wont ever get Spotlights with volumetrics and shadows enabled?

**Comment by Zireael07** [comment]
closed as duplicate - read the comment above instead of just reading the automated message

**Comment by Xyotic** [comment]
It did just confuse me. Thats all
